src/nvim/options.lua

```diff
@@ -9029,6 +9029,24 @@ return {
       type = 'string',
       varname = 'p_titlestring',
     },
+    {
+      abbreviation = 'tsml',
+      defaults = { if_true = 256 },
+      desc = [=[
+        This option determines the maximum number of treesitter nodes matched
+        when a cursor query is executed. Higher values allow for deeper matches
+        at the expense of query execution time, while lower values can be used
+        to speed up the execution time.
+
+        The default value is 256. Maximum is 64000 and minimum is 32.
+      ]=],
+      full_name = 'tsmatchlimit',
+      scope = { 'global', 'buffer' },
+      short_desc = N_('maximum number of nodes in treesitter matches for each query'),
+      type = 'number',
+      redraw = { 'ui_option', 'current_buffer' },
```

Comment:
It doesn't make sense for this to be a UI option.

Reply:
I don't really know these options that well and just kept `'ui-option'` from #16008. 

Do you think it should just be `'current_buffer'`? Or do I need to change something else here?

Reply:
Just `current_buffer`

Reply:
I think you should wait, honestly, like I advised above.

Reply:
> I think you should wait, honestly, like I advised above.

I am not sure if this was for me or not, but for such small stuff I will probably just update it if people comment on it. But I still don't plan to make any major edits until there is some agreement about how to do this.

Reply:
The point is that we may not want to go with this approach _at all_ (which in fact I don't think we do, see below).